### DISCUSSION TITLE
How can we unit test TextChanged and Validator event if its in a column of a DataGrid

### DISCUSSION THREAD

**Discussion by meekhasaji** [Megabit/Blazorise#2797] [Q&A] [answered]
Hi,
I am using Bunit , MSTests for testing the components.
this TextEdit actually is under such a structure

```
<DataGridColumn TItem="Model" Field="@nameof(Model.Value)" Caption="Value" Sortable="true" Editable="true">
                       <EditTemplate>
                           <Validation Validator="ValidateFunction">
                               <TextEdit Text="@((string) context.CellValue)"
                                         TextChanged="@(value => context.CellValue = value)">
                                   <Feedback>
                                       <ValidationError />
                                   </Feedback>
                               </TextEdit>
                           </Validation>
                       </EditTemplate>
                   </DataGridColumn>
```
They (TextEdit and the Validation ) do not come up in the Markup content when we try to render the component containing this DataGrid in a test case.
I'm a bit of a web novice, so any suggestions are gratefully accepted. 
Thanks in advance,

Meekha

**Comment by David-Moreira** [comment, marked as answer]
Hello @meekhasaji,
Could you send us your test sample? Have you forgotten to load some data into the DataGrid? You have to simulate every step you would do in the browser, but with Bunit.
